src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -335,6 +349,52 @@ function attachInteropInvoker(): void {
   });
 }
 
+async function loadTimezone(timeZoneResource: LoadingResource) : Promise<void> {
+  const runDependencyId = `blazor:timezonedata`;
+  Module.addRunDependency(runDependencyId);
+  const request = await timeZoneResource.response;
+  let arrayBuffer = await request.arrayBuffer();
+
+  // The contents of the TZ file look like so
+  //
+  // [4 - byte length of manifest]
+  // [json manifest]
+  // [data bytes]
+  //
+  // The json manifest is an array that looks like so:
+  //
+  // [...["America/Fort_Nelson",2249],["America/Glace_Bay",2206]..]
+  //
+  // where the first token in each array is the relative path of the file on disk, and the second is the
+  // length of the file. The starting offset of a file can be calculated using the lengths of all files
+  // that appear prior to it.
+
+  const manifestSize = readInt32LE(new Uint8Array(arrayBuffer.slice(0, 4)), 0);
+  arrayBuffer = arrayBuffer.slice(4);
+  const manifestContent = new TextDecoder().decode(arrayBuffer.slice(0, manifestSize));
```

Comment:
Note that TextDecoder isn't available on Edge older than January 2020: https://caniuse.com/#feat=textencoder

We already have a polyfill in our sources here though - see `src\Rendering\RenderBatch\Utf8Decoder.ts`. This polyfill uses `TextDecoder` when available, and falls back to a JS implementation if not.

It's unfortunate to make `blazor.webassembly.js` bigger by referencing that, but probably is worth it as some customers may still be on Edge that's more than a few months old, and this would be the only incompatible thing for them.

Reply:
January was the first [public release](https://www.zdnet.com/article/microsofts-chromium-based-edge-browser-to-be-generally-available-january-15-2020) of Edge. My guess is caniuse does not track preview builds of the browser.

Reply:
Huh? Edge has existed in pre-Chromium form for many years, and that pre-Chromium version does support WebAssembly.

Reply:
Oops, TIL